### PR TITLE
feat: MEMBER

### PR DESCRIPTION
**Pull request by SoA-Lee**
### 변경사항
**AS-IS**
1. 회원가입/로그인/내 정보 
   - Member Entity 칼럼 추가(Id&Email 분리)에 따라 코드 수정
      + Github Action 설정 전, 회원가입 및 로그인 기능을 구현하여 merge한 상태입니다. 참고 부탁드립니다.

2. 기타
   - Error Code 및 예외처리 작성

**TO-BE**
1. 기능 구현
   - 탈퇴하기
   - 상품 관련 Entity 작성하기
   - 상품 등록 및 수정하기

2. 테스트코드 작성
   - 회원가입 / 로그인 테스트 코드 작성하기

### 테스트
- [x] 테스트 코드
   - 회원 레포지토리 test
   - 이메일 인증 test
- [ ] API 테스트

### CODE REVIEW

**Comment by SoA-Lee** [comment]
|File|Coverage [10.22%]|:x:|
|:-|:-:|:-:|
|[Member.java](https://github.com/SoA-Lee/N-BUY/blob/b41b2535565db7c1b56bf405ec6b8b37c9800bd5/src%2Fmain%2Fjava%2Fcom%2Fdalgorithm%2Fnbuy%2Fmember%2Fentity%2FMember.java)|37.85%|:x:|
|[MemberServiceImpl.java](https://github.com/SoA-Lee/N-BUY/blob/b41b2535565db7c1b56bf405ec6b8b37c9800bd5/src%2Fmain%2Fjava%2Fcom%2Fdalgorithm%2Fnbuy%2Fmember%2Fservice%2Fimpl%2FMemberServiceImpl.java)|15.09%|:x:|
|[MemberDto.java](https://github.com/SoA-Lee/N-BUY/blob/b41b2535565db7c1b56bf405ec6b8b37c9800bd5/src%2Fmain%2Fjava%2Fcom%2Fdalgorithm%2Fnbuy%2Fmember%2Fdto%2FMemberDto.java)|0%|:x:|
|[NbuyException.java](https://github.com/SoA-Lee/N-BUY/blob/b41b2535565db7c1b56bf405ec6b8b37c9800bd5/src%2Fmain%2Fjava%2Fcom%2Fdalgorithm%2Fnbuy%2Fcommon%2FException%2FNbuyException.java)|0%|:x:|
|[MemberInput.java](https://github.com/SoA-Lee/N-BUY/blob/b41b2535565db7c1b56bf405ec6b8b37c9800bd5/src%2Fmain%2Fjava%2Fcom%2Fdalgorithm%2Fnbuy%2Fmember%2Fmodel%2FMemberInput.java)|0%|:x:|
|[ErrorCode.java](https://github.com/SoA-Lee/N-BUY/blob/b41b2535565db7c1b56bf405ec6b8b37c9800bd5/src%2Fmain%2Fjava%2Fcom%2Fdalgorithm%2Fnbuy%2Fcommon%2Ftype%2FErrorCode.java)|0%|:x:|
|[MemberController.java](https://github.com/SoA-Lee/N-BUY/blob/b41b2535565db7c1b56bf405ec6b8b37c9800bd5/src%2Fmain%2Fjava%2Fcom%2Fdalgorithm%2Fnbuy%2Fmember%2Fcontroller%2FMemberController.java)|0%|:x:|

|Total Project Coverage|8.64%|:x:|
|:-|:-:|:-:|

File: src/main/java/com/dalgorithm/nbuy/common/Exception/NbuyException.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+import com.dalgorithm.nbuy.common.type.ErrorCode;
+import lombok.*;
+
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+public class NbuyException extends RuntimeException {
```
Comment (spring-kotlin-lover) [inline src/main/java/com/dalgorithm/nbuy/common/Exception/NbuyException.java]: NbuyException 에 대한 네이밍을 명확히 해보시는 것은 어떨까요? 개발자로써 가장 신경써야되는 부분이 네이밍이라고도 생각됩니다.

https://tecoble.techcourse.co.kr/post/2020-04-26-Method-Naming/

File: src/main/java/com/dalgorithm/nbuy/member/service/impl/MemberServiceImpl.java [hunk 2]
```diff
@@ -36,18 +36,16 @@ public boolean register(MemberInput memberInput) {
 
         Optional<Member> optionalMember = memberRepository.findById(memberInput.getUserId());
         if (optionalMember.isPresent()) {
-            //현재 userId에 해당하는 데이터 존재
-            return false;
+            throw new NbuyException(ErrorCode.MEMBER_ALREADY_EXIST);
```
Comment (spring-kotlin-lover) [inline src/main/java/com/dalgorithm/nbuy/member/service/impl/MemberServiceImpl.java]: 위에서 언급드렸듯이 NbuyException 과 에러코드인 Member_ALREADY_EXISTS 가 어떤의미인지 정확하게 연결이 어려운 것 같습니다.

패키지명이 nBuy 라고 해서 꼭 따라갈 필요는 없을 것도 같아요.

오히려 Domain 에 대한 예외라고 해서 DomainException 과 같은 추상화 레벨로 맞추는 것이 에러코드와도 쉽게 연결될 수도 있습니다.

File: src/test/java/com/dalgorithm/nbuy/member/service/impl/MemberServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+    @Mock
+    private MemberRepository memberRepository;
+
+    @InjectMocks
+    private MemberServiceImpl memberService;
+
+    @Test
+    @DisplayName("이메일 인증 성공")
+    void emailAuthTest() {
```
Comment (spring-kotlin-lover) [inline src/test/java/com/dalgorithm/nbuy/member/service/impl/MemberServiceImplTest.java:34]: 작성해주신 테스트가 happy case 만 있네요 조금 더 다양하게 테스트해보시는 것은 어떨까요?
테스트를 작성한다는 것은 어떤 상황을 생각한 기대값들을 검증해야되는데 happy case 하나만으로는 테스트를 작성하는 의미가 많이 사라지게 됩니다.

**Review by spring-kotlin-lover — COMMENTED** [review]
안녕하세요 :) 저는 무지멘토라고 하구요~! 리뷰드릴만한 포인트들 코멘트 드렸습니다.

File: src/test/java/com/dalgorithm/nbuy/member/service/impl/MemberServiceImplTest.java (hunk 3 above)
Comment (SoA-Lee) [reply src/test/java/com/dalgorithm/nbuy/member/service/impl/MemberServiceImplTest.java:34]: 네 리뷰해주신 내용 토대로 다시 수정해보겠습니다! 감사합니다.

**Comment by SoA-Lee** [comment]
|File|Coverage [16.79%]|:x:|
|:-|:-:|:-:|
|[ErrorCode.java](https://github.com/SoA-Lee/N-BUY/blob/e6d7a6e6ebb44d2b14b5a272daece26ddf6d0122/src%2Fmain%2Fjava%2Fcom%2Fdalgorithm%2Fnbuy%2Fexception%2FErrorCode.java)|97.44%|:green_apple:|
|[Member.java](https://github.com/SoA-Lee/N-BUY/blob/e6d7a6e6ebb44d2b14b5a272daece26ddf6d0122/src%2Fmain%2Fjava%2Fcom%2Fdalgorithm%2Fnbuy%2Fmember%2Fentity%2FMember.java)|39.27%|:x:|
|[MemberServiceImpl.java](https://github.com/SoA-Lee/N-BUY/blob/e6d7a6e6ebb44d2b14b5a272daece26ddf6d0122/src%2Fmain%2Fjava%2Fcom%2Fdalgorithm%2Fnbuy%2Fmember%2Fservice%2Fimpl%2FMemberServiceImpl.java)|18%|:x:|
|[DomainException.java](https://github.com/SoA-Lee/N-BUY/blob/e6d7a6e6ebb44d2b14b5a272daece26ddf6d0122/src%2Fmain%2Fjava%2Fcom%2Fdalgorithm%2Fnbuy%2Fexception%2FDomainException.java)|16.67%|:x:|
|[MemberDto.java](https://github.com/SoA-Lee/N-BUY/blob/e6d7a6e6ebb44d2b14b5a272daece26ddf6d0122/src%2Fmain%2Fjava%2Fcom%2Fdalgorithm%2Fnbuy%2Fmember%2Fdto%2FMemberDto.java)|0%|:x:|
|[MemberController.java](https://github.com/SoA-Lee/N-BUY/blob/e6d7a6e6ebb44d2b14b5a272daece26ddf6d0122/src%2Fmain%2Fjava%2Fcom%2Fdalgorithm%2Fnbuy%2Fmember%2Fcontroller%2FMemberController.java)|0%|:x:|
|[ErrorResponse.java](https://github.com/SoA-Lee/N-BUY/blob/e6d7a6e6ebb44d2b14b5a272daece26ddf6d0122/src%2Fmain%2Fjava%2Fcom%2Fdalgorithm%2Fnbuy%2Fexception%2FErrorResponse.java)|0%|:x:|
|[GlobalExceptionHandler.java](https://github.com/SoA-Lee/N-BUY/blob/e6d7a6e6ebb44d2b14b5a272daece26ddf6d0122/src%2Fmain%2Fjava%2Fcom%2Fdalgorithm%2Fnbuy%2Fexception%2FGlobalExceptionHandler.java)|0%|:x:|
|[MemberInput.java](https://github.com/SoA-Lee/N-BUY/blob/e6d7a6e6ebb44d2b14b5a272daece26ddf6d0122/src%2Fmain%2Fjava%2Fcom%2Fdalgorithm%2Fnbuy%2Fmember%2Fmodel%2FMemberInput.java)|0%|:x:|

|Total Project Coverage|15.19%|:x:|
|:-|:-:|:-:|

**Pull request merged**